main/entry.c

```diff
@@ -635,6 +640,21 @@ static size_t appendSourceLine ( void putc_func (char , void *), const char *con
 		putc_func (c, data);
 		++length;
 	}
+
+	if (length > Option.patternLengthLimit)
+	{
+		char lenpat[32];
+		char *l;
+		int r;
+
+		r = snprintf (lenpat, 32, "\\.\\{%lu}",
```

Comment:
Now I understand what you were trying to do using patterns in the tags file.
As a KIS(S) alternative, have you considered simply omitting the remainder of the line and dropping `$` instead of adding this complex pattern?